### ISSUE TITLE
Fails to even load in VS2022

### ISSUE THREAD

**Issue by smaillet-ms** [loop8ack/ExtensionPackTools#82]
I downloaded this from the VS marketplace (https://marketplace.visualstudio.com/_apis/public/gallery/publishers/Loop8ack/vsextensions/ExtensionManager2022/1.2.182/vspackage)
And restarted VS, and immediately got an error about something not working and I should check ActivityLog.xml (This is, of course a terrible UX for VS as it should be ablee to tell me more, or at the very least Link to a properly rendered version of the file!) But I digress...

The activity log contained the following:
```
SetSite failed for package [VsPackage]Source: 'ExtensionManager2022' Description: Not supported visual studio version: 17.8.34601.278
System.InvalidOperationException: Not supported visual studio version: 17.8.34601.278
   at ExtensionManager.VisualStudioServiceFactory.<CreateAsync>d__0.MoveNext() in C:\projects\extensionpacktools\src\VS2022\VisualStudioServiceFactory.cs:line 18
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at ExtensionManager.VsPackage.<InitializeAsync>d__0.MoveNext() in C:\projects\extensionpacktools\src\ExtensionManager.Shared\VsPackage.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Shell.AsyncPackage.<>c__DisplayClass21_0.<<Microsoft-VisualStudio-Shell-Interop-IAsyncLoadablePackageInitialize-Initialize>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.Services.VsTask.RethrowException(AggregateException e)
   at Microsoft.VisualStudio.Services.VsTask.InternalGetResult(Boolean ignoreUIThreadCheck)
```
How can it be complaining about an unsupported VS version when the VSIX package correctly installed it. Does this support VS2022 or not?

**Comment by MagicAndre1981** [comment]
Looks like only non LTSC versions are supported as he checks for [17.9](https://github.com/loop8ack/ExtensionPackTools/blob/d37097cc870eee39fa5d523420db77d3a37902e4/src/VS2022/VisualStudioServiceFactory.cs#L15) (which already has [buildin feature to export/import extensions via Installer](https://devblogs.microsoft.com/visualstudio/visual-studio-2022-17-9-now-available/#extensibility))

**Comment by loop8ack** [comment]
Correct - currently, only the current version 17.9 or the preview version 17.10 is supported.

Unfortunately, this is necessary because this extension depends on internal DLLs of Visual Studio, which often change and then have to be manually updated.

I am currently working on an experiment that should be able to dynamically load these DLLs and generate the necessary code. However, this process might take some time to ensure everything works correctly.

Until then, the only solution is for you to update Visual Studio.

**Comment by MagicAndre1981** [comment]
update the[ min version to 17.9](https://github.com/loop8ack/ExtensionPackTools/blob/e17e56d01633b6e63008bec0a61414988fcc2494/src/VS2022/source.extension.vsixmanifest#L14) in the manifest

**Comment by loop8ack** [comment]
Good point - I hadn't thought of that, thanks.
I'll do it straight away.

**Comment by smaillet-ms** [comment]
Yes, please update the manifest to reflect current requirements. Updating isn't an option as we are locked to the LTSC for 17.8 until VS team deals with egregious bugs that wreck the UX. Though If the feature is built into the installer for 17.9 I have to wonder - what's the point of this extension?

**Comment by MagicAndre1981** [comment]
the point of the extension is that is this feature was not available at creation time and if you use an older LTSC and want to try a newer VS version if it fixes some bugs, the export of extension via config file is likely not available.

**Comment by loop8ack** [comment]
I think that this extension still has a right to exist, as it is better integrated and offers more functions. In addition, the installer only seems to be able to export a limited number of extensions.

The restriction to the current Visual Studio version is of course a disadvantage, but I'm already working on that.

My progress is also good, I was able to carry out the first successful tests today. However, I will need a few more days.

**Comment by loop8ack** [comment]
@smaillet-ms

I just released a new version.
It includes a new architecture as well as minor improvements. Of course, it also includes dynamic loading of the Visual Studio DLLs. 
This version is still somewhat experimental, and I look forward to feedback.

Please test if it works with your older version and keep me updated.

**Comment by kkm000** [comment]
@loop8ack, it would be amazing, if it's not too hard, to mention related issues within commit messages. It sometimes takes just some eyeballing to understand the problem, especially when a feature is experimental. This improves chances of someone contributing a fix.

BTW, unless I'm asking something that I shouldn't (in this case please just say that I shouldn't), why did Mads abandon this project?

**Comment by loop8ack** [comment]
@kkm000 
I have released a comprehensive rebuild of the project, so I cannot limit this issue to a specific commit.
But I will do that for future issues.

> why did Mads abandon this project

It started with exactly the same problem, the extension stopped working with Visual Studio 2022 Preview due to an incompatibility of different DLL versions, for which I then developed the temporary solution. I asked him if he was still maintaining the project, which he denied. I then asked if I could take it over - and here we are.

**Comment by kkm000** [comment]
@loop8ack Thank you for stepping in, really appreciate it!

**Comment by TwentyFourMinutes** [comment]
> I just released a new version.
It includes a new architecture as well as minor improvements. Of course, it also includes dynamic loading of the Visual Studio DLLs.
This version is still somewhat experimental, and I look forward to feedback.

@loop8ack Thank you really much for your work. Can confirm the extension working again after updating to your fix (17.10.0 Preview 3.0).

**Comment by loop8ack** [comment]
@TwentyFourMinutes
Glad it's working for you! Feel free to report any other issues or suggestions.

Thanks for the feedback!

@smaillet-ms 
Given the update (1.3.1) and lack of further feedback, I'm closing this issue.
If the problem persists or you have more questions, feel free to reopen or create a new issue.

Thanks!
